02stl/cpp/snippets.cc

```diff
@@ -0,0 +1,56 @@
+/// viter
+std::vector<int> myVector = {1,2,3,4};
+for (int i=0; i<myVector.size(); ++i) {
+    std::cout << myVector[i] << std::endl;
+}
+
+
+///liter
+#include <iterator>
+
+std::list<int> mylist = {1,2,3,4};
+std::list<int>::iterator it=mylist.begin()
+for ( ; it!=mylist.end(); ++it) {
+    std::cout << *it << std::endl;
+}
+
+    
+/// pairs
+#include <utility>      // std::pair
+
+int main () {
+  std::pair <int,int> foo;                                     // default constructor
```

Comment:
Declaring variables way before their use is bad style.

Better to do `auto foo = ??` at place of first instantiation.
In practice, if something is bad style and could be picked up by the students, we shouldn't show it to them.
Better to just *say* `there is a default constructor that takes no arguments, but it is good practice to always initialize variables`, and not show the default constructor in the notes.